idelib/dataset.py

```diff
@@ -254,6 +256,8 @@ def __init__(self, stream, name=None, exitCondition=None, quiet=True):
         self.loadCancelled = False
         self.loading = True
         self.filename = getattr(stream, "name", None)
+
+        self._channelDataLock = Lock()
```

Comment:
It seems like this could/should be used in more places than `fillCache()` and `_accessCache()` (e.g., `EventArray.append()` and various value-getting methods), but that's beyond the scope of this particular PR.

Reply:
I agree that it should go in append, it was very easy to add so I did.